community/rfcs/24-08-02-OPEA-AIAvatarChatbot.md

```diff
@@ -0,0 +1,140 @@
+                direction TB
+                C((ASR<br>3001)) -. Post .-> D{{whisper-service<br>7066}}
+                E((LLM<br>3007)) -. Post .-> F{{tgi-service<br>3006}}
+                G((TTS<br>3002)) -. Post .-> H{{speecht5-service<br>7055}}
+            end
+            subgraph AvatarAnimation["Avatar Animation"]
+                direction TB
+                I((AvatarAnime<br>3010)) -. Post .-> J{{animation<br>3009}}
+            end
+            AudioQnA ==> AvatarAnimation
+        end
+        Megaservice --> |Output| K[Response]
+    end
+
+    subgraph Legend
+        direction LR
+        L([Microservice]) ==> M([Microservice])
+        N([Microservice]) -.-> O{{Server API}}
+    end
+```
+The AvatarChatbot megaservice is a new service that integrates the existing AudioQnA service with the new animation microservice. The AudioQnA service is a pipeline that takes user audio input, converts it to text, generates an LLM response, and converts the response to audio output. The animation microservice is a new service that takes the audio response from the AudioQnA service, generates an animated avatar response, and sends it back to the user. The AvatarChatbot Gateway invokes the AvatarChatbot backend megaservice to generate the response.
+
+#### animation microservice
+The animation microservice is a new service that generates animated avatar videos from audio and image/video inputs. The animation microservice takes the audio response from the AudioQnA service, synchronizes the audio response mel-spectrogram chunks with image/video frames, and generates a high-quality video of the avatar speaking in real-time. The animation microservice currently uses the [Wav2Lip-GFPGAN](https://github.com/ajay-sainy/Wav2Lip-GFPGAN) model for lip synchronization and face animation.  
+
+User can build their own Docker image with `Dockerfile_hpu` and create a Docker container on Gaudi2 instance to run the animation microservice. They can then validate the service by sending a POST request to the server API, while providing audio and image/video inputs. The animation microservice will generate an animated avatar video response and save it to the specified output path.
+
+Support for alternative SoTA models such as [SadTalker](https://github.com/OpenTalker/SadTalker) and [LivePortrait](https://github.com/KwaiVGI/LivePortraitare) in progress.
+
+#### AvatarChatbot megaservice
+The AvatarChatbot megaservice is a new service that integrates the existing microservices that comprise AudioQnA service with the new animation microservice. The AudioQnA service is a pipeline that takes user audio input, converts it to text, generates an LLM response, and converts the response to audio output. The animation microservice is a new service that takes the audio response from the AudioQnA service, generates an animated avatar response, and sends it back to the user. The megaflow is as follows:  
+asr -> llm -> tts -> animation
```

Comment:
might be use mermaid flowchart here instead?

Reply:
Included a more accurate mermaid flowchart for the new `AvatarChatbot` megaservice.